src/utils/formatSearchResponse.ts

```diff
@@ -17,20 +18,17 @@ const formatSearchResponse = async (
     const startDate: string = searchItem?._source?.resourceTemporalExtentDetails?.[0]?.start?.date ?? '';
     const endDate: string = searchItem?._source?.resourceTemporalExtentDetails?.[0]?.end?.date ?? '';
     const studyPeriod = getStudyPeriod(startDate, endDate);
+    const publishedBy = getOrganisationDetails(searchItem?._source?.contactForResource ?? []);
+    const organisationDetails = getOrganisationDetails(searchItem?._source?.contactForResource ?? [], true);
+
     const item: ISearchItem = {
       id: searchItem?._id,
       title: searchItem?._source?.resourceTitleObject?.default ?? '',
-      publishedBy: searchItem?._source?.OrgObject?.default ?? '',
+      publishedBy: publishedBy.organisationValue,
       content: searchItem?._source?.resourceAbstractObject?.default ?? '',
       studyPeriod,
       resourceLocator: searchItem?._source?.resourceIdentifier?.[0]?.codeSpace ?? '',
-      ncea_catalogue_number: searchItem?._source?.uuid,
-      host_catalogue_number: `${searchItem?._source?.resourceIdentifier?.[0]?.codeSpace ?? ''} ${searchItem?._source?.resourceIdentifier?.[0]?.code ?? ''}`,
-      // Keeping this as a placeholder, as the Coupled Resource is not available now
-      host_catalogue_entry: '',
-      resource_type_and_hierarchy: searchItem?._source?.resourceType?.[0] ?? '',
-      hierarchy_level: searchItem?._source?.cl_hierarchyLevel?.[0]?.default ?? '',
-      resource_locators: `${searchItem?._source?.cl_function?.[0]?.default} from ${searchItem?._source?.link?.[0]?.nameObject?.default} (<a class="govuk-link" href="${searchItem?._source?.link?.[0]?.urlObject?.default}" target="_blank">${searchItem?._source?.link?.[0]?.urlObject?.default}</a>)`,
+      organisationName: organisationDetails.organisationValue,
     };
     if (isDetails) {
       getOtherDetails(item, searchItem);
```

Comment:
```suggestion
      const otherDetails = getOtherDetails(searchItem);
      items = {...items, ...otherDetails};
 
```

getOtherDetails needs to return the object. playing with object referencing would be hard to debug if these functions used inside a loop construct.

Reply:
Updated